### DISCUSSION TITLE
Custom mask tokens

### DISCUSSION THREAD

**Discussion by mncoldsmoke** [quasarframework/quasar#12759] [General - Components / Directives / etc] [answered]
I've looked at the existing tokens in use-masks.js, and it seems straight forward how they work. I would like to add my own custom tokens for example alphanumeric plus dash (-) or limiting to state-abbreviation first letters / second letters.

My issue is that TOKENS isn't exported and I can't tell if there is a way to access it to add some tokens.

Is there a spot that I'm missing where the TOKENS const is exposed so I can extend it?

If not how do I request this as an update?

**Comment by symphograph** [comment]
I'm interested too.

**Comment by rstoenescu** [comment, marked as answer]
New QInput prop (mask-tokens) will be available in Quasar v2.18.4
